src/components/PupilComponents/QuizUtils/questionUtils.ts

```diff
@@ -0,0 +1,8 @@
+import { LessonOverviewQuizData } from "@/node-lib/curriculum-api-2023/shared.schema";
+
+export const getInteractiveQuestions = (questions: LessonOverviewQuizData) =>
+  questions?.length
+    ? questions.filter(
+        (question) => question.questionType !== "explanatory-text",
+      )
+    : [];
```

Comment:
🏄🏼 normally i'd argue that the questions should be non-nullable and that the value should be coalesced to an array before we get here. That would mean we don't have to deal with the "no questions" case since the type system would have us covered. However I know that sonar cloud would complain about that line only being partially covered 😮‍💨